File: 3154-maximum-value-of-an-ordered-triplet-i/3154-maximum-value-of-an-ordered-triplet-i.cpp
class Solution {
public:
    long long maximumTripletValue(vector<int>& nums) {
        // watched from editorial -> 
        // we want that the first thing should be maximized nums[i]-nums[j] 
        int currMaxnum=0;
        int currMaxdiff=0;
        long long ans=0;
        for(int k=0;k<nums.size();k++){
            ans=max((long long)nums[k]*currMaxdiff,ans);
            currMaxnum=max(nums[k],currMaxnum);
            currMaxdiff=max(currMaxdiff,currMaxnum-nums[k]);
        }

        return ans;
    }
};

File: 3154-maximum-value-of-an-ordered-triplet-i/3154-maximum-value-of-an-ordered-triplet-i.cs
public class Solution {
    public long MaximumTripletValue(int[] nums) {
        long ans=0;
        int n=nums.Length;

        for(int i=0;i<n;i++){
            for(int j=i+1;j<n;j++){
                for(int k=j+1;k<n;k++){
                    long check=(nums[i] - nums[j]);
                    check=check*(nums[k]);
                    if(check>0){
                        ans=Math.Max(ans,check);
                    }
                }
            }
        }
    return ans;
    }
}